dev/com.ibm.ws.jca.cm/src/com/ibm/ejs/j2c/MCWrapper.java

```diff
@@ -726,7 +729,30 @@ protected void setConnectionManager(ConnectionManager cm) {
             throw e;
         }
 
-        recoveryToken = cm.getRecoveryToken();
+        if (qmidenabled) {
```

Comment:
There ought to be a way of optimizing out of having to attempt this reflection every time for JDBC path.  Possibly by initializing,
qmidenabled = !(mc instanceof WSManagedConnection);
in the constructor?  Or there are probably better ways if you explore around with what is available.

Reply:
Still thinking about this one, but to avoid the reflection, it does not hurt to add qmidenabled = !(mc instanceof WSManagedConnection);